Add dual pattern matching to kgrep/dgrep with --ne and --default flags

rahulduvedi opened the pull request:
Enhanced `kgrep` and `dgrep` Commands

### Check for success AND absence of errors
`kgrep pod/myapp -e "server started" --ne "error" --ne "timeout" --ne "panic"`

### defaults:  common error patterns  
`kgrep pod/myapp -e "ready" --std-errors`

### Flexible exclusions: Use defaults but skip specific patterns
`kgrep pod/myapp --std-errors --ne-exclude "timeout" --ne "custom_error"`

Why --invert-match Isn't Sufficient
The existing `--invert-match` flag has critical limitations:
Binary Choice Problem: Can only validate one thing at a time, either expected patterns OR unexpected patterns, never both simultaneously.

Global Inversion: Flips the logic of ALL patterns, making -e patterns behave opposite to their name, creating
confusion.

This Solution: Mixed validation allows checking for required patterns AND forbidden patterns in the same command, matching real-world validation needs.

Usage Examples

### Basic: Verify startup without common errors
`kgrep deploy/app -e "ready" --std-errors`

### Advanced: Custom error patterns with exclusions  
`kgrep deploy/database -e "connection established" --std-errors --ne-exclude "expected error"`

rakshitgondwal commented on tw/pkg/commands/dgrep/dgrep.go:
```diff
@@ -55,7 +76,10 @@ func Command() *cobra.Command {
+	cmd.Flags().StringArrayVarP(&cfg.Patterns, "regexp", "e", nil, "regular expression to match (must be present)")
+	cmd.Flags().StringArrayVar(&cfg.NotExpected, "ne", nil, "regular expression that must NOT be present")
+	cmd.Flags().StringArrayVar(&cfg.NotExpectedExclude, "ne-exclude", nil, "exclude specific patterns from --default (only works with --default)")
+	cmd.Flags().BoolVar(&cfg.DefaultErrors, "default", false, "add default error patterns (adds 22 common error patterns)")
```
```suggestion
	cmd.Flags().BoolVar(&cfg.DefaultErrors, "default", false, "add default error patterns (adds 13 common error patterns)")
```

Dentrax replied:
Let's remove that number or dynamically calculate using `len()`.

Dentrax commented on tw/pkg/commands/dgrep/dgrep.go:
```diff
@@ -24,16 +24,37 @@ const (
 )
 
+// Common error patterns for --all flag
+var commonErrorPatterns = []string{
+	"ERROR",
+	"Error", 
+	"error",
+	"Fatal",
+	"FATAL",
+	"fatal",
+	"Fail",
+	"FAIL", 
+	"fail",
+	"Exception",
+	"exception",
+	"panic",
+	"PANIC",
+}
+
```
Same here https://github.com/chainguard-dev/tw/pull/180#discussion_r2314258222

Dentrax commented on tw/pkg/commands/kgrep/kgrep.go:
```diff
@@ -60,7 +81,10 @@ func Command() *cobra.Command {
+	cmd.Flags().StringArrayVarP(&cfg.Patterns, "regexp", "e", nil, "regular expression to match that must be present")
+	cmd.Flags().StringArrayVar(&cfg.NotExpected, "ne", nil, "regular expression that must NOT be present")
+	cmd.Flags().StringArrayVar(&cfg.NotExpectedExclude, "ne-exclude", nil, "exclude specific patterns from --default (only works with --default)")
+	cmd.Flags().BoolVar(&cfg.DefaultErrors, "default", false, "add default error patterns (adds 22 common error patterns)")
```
Same as https://github.com/chainguard-dev/tw/pull/180#discussion_r2314258928

rahulduvedi commented:
@rakshitgondwal @Dentrax
Resolved all the comments, here is the summary. PTAL
Thank you

Removed duplicate patterns
Dynamic pattern count (Help text shows "13" instead of hardcoded "22")
Added production patterns: Traceback, NullPointerException, segmentation fault
 Updated `--ne-exclude` to use valid patterns

Dentrax commented on tw/pkg/commands/dgrep/dgrep.go:
```diff
@@ -24,16 +25,37 @@ const (
 
+// Common error patterns for --default flag
+var commonErrorPatterns = []string{
+	"ERROR",
+	"FATAL",
+	"FAIL",
+	"Exception",
+	"panic",
+	"Traceback",
+	"command not found",
+	"NullPointerException",
+	"RuntimeException",
+	"TimeoutException",
+	"OutOfMemoryError",
+	"StackOverflow",
+	"segmentation fault",
```
Same for `kgrep`:

```suggestion
	"ERROR",
	"FATAL",
	"FAIL",
	"Exception",
	"panic",
	"Traceback.*most.recent.call",
	"command not found",
	"java.lang.*Exception",
	"segmentation fault",
	"org.jruby.exceptions",
	"Gem::MissingSpecError"
```
